### PR TITLE
Fix client generators to work with single war

### PR DESCRIPTION
**Pull request by jaceko**

### CODE REVIEW

File: generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java [hunk 1]
```diff
@@ -58,9 +58,9 @@ public void shouldGenerateClassWithAnnotations() throws Exception {
                         .build(),
                 configurationWithBasePackage(BASE_PACKAGE, outputFolder, generatorProperties().withServiceComponentOf("COMMAND_API")));
 
-        Class<?> generatedClass = compiler.compiledClassOf(BASE_PACKAGE, "RemoteEventProcessorMessageContextCakeshopControllerCommand");
+        Class<?> generatedClass = compiler.compiledClassOf(BASE_PACKAGE, "RemoteCommandApi2EventProcessorMessageContextCakeshopControllerCommand");
```
Comment (skiddykong) [inline generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java]: final

File: generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java [hunk 2]
```diff
@@ -77,7 +77,7 @@ public void shouldGenerateClassWithCommandControllerAnnotation() throws Exceptio
                         .build(),
                 configurationWithBasePackage(BASE_PACKAGE, outputFolder, generatorProperties().withServiceComponentOf("COMMAND_CONTROLLER")));
 
-        Class<?> generatedClass = compiler.compiledClassOf(BASE_PACKAGE, "RemoteEventProcessorMessageContextCakeshopHandlerCommand");
+        Class<?> generatedClass = compiler.compiledClassOf(BASE_PACKAGE, "RemoteCommandController2EventProcessorMessageContextCakeshopHandlerCommand");
```
Comment (skiddykong) [inline generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java]: final

File: generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java [hunk 3]
```diff
@@ -92,7 +92,7 @@ public void shouldGenerateClientForEventTopic() throws Exception {
                         .build(),
                 configurationWithBasePackage(BASE_PACKAGE, outputFolder, generatorProperties().withServiceComponentOf("EVENT_PROCESSOR")));
 
-        Class<?> generatedClass = compiler.compiledClassOf(BASE_PACKAGE, "RemoteEventProcessorMessageContextPublicEvent");
+        Class<?> generatedClass = compiler.compiledClassOf(BASE_PACKAGE, "RemoteEventProcessor2EventProcessorMessageContextPublicEvent");
```
Comment (skiddykong) [inline generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java]: final

File: generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java [hunk 4]
```diff
@@ -105,9 +105,9 @@ public void shouldCreateLoggerConstant() throws Exception {
                                 .withRelativeUri("/cakeshop.controller.command")
                                 .withDefaultPostAction())
                         .build(),
-                configurationWithBasePackage(BASE_PACKAGE, outputFolder, generatorProperties().withDefaultServiceComponent()));
+                configurationWithBasePackage(BASE_PACKAGE, outputFolder, generatorProperties().withServiceComponentOf("COMMAND_CONTROLLER")));
 
-        Class<?> generatedClass = compiler.compiledClassOf(BASE_PACKAGE, "RemoteEventProcessorMessageContextCakeshopControllerCommand");
+        Class<?> generatedClass = compiler.compiledClassOf(BASE_PACKAGE, "RemoteCommandController2EventProcessorMessageContextCakeshopControllerCommand");
```
Comment (skiddykong) [inline generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java]: final

File: generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java [hunk 5]
```diff
@@ -125,9 +125,10 @@ public void shouldCreateJmsEnvelopeSenderVariable() throws Exception {
...
```
Comment (skiddykong) [inline generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java]: final

File: generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java [hunk 6]
```diff
@@ -145,9 +146,9 @@ public void shouldGenerateMethodAnnotatedWithHandlesAnnotation() throws Exceptio
                                 .withRelativeUri("/cakeshop.controller.command")
                                 .with(httpActionWithDefaultMapping(POST, "application/vnd.cakeshop.actionabc+json")))
                         .build(),
-                configurationWithBasePackage(BASE_PACKAGE, outputFolder, generatorProperties().withDefaultServiceComponent()));
+                configurationWithBasePackage(BASE_PACKAGE, outputFolder, generatorProperties().withServiceComponentOf("COMMAND_CONTROLLER")));
 
-        Class<?> generatedClass = compiler.compiledClassOf(BASE_PACKAGE, "RemoteEventProcessorMessageContextCakeshopControllerCommand");
+        Class<?> generatedClass = compiler.compiledClassOf(BASE_PACKAGE, "RemoteCommandController2EventProcessorMessageContextCakeshopControllerCommand");
```
Comment (skiddykong) [inline generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java]: final

File: generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java [hunk 7]
```diff
@@ -145,9 +146,9 @@ public void shouldGenerateMethodAnnotatedWithHandlesAnnotation() throws Exceptio
...
 
         List<Method> methods = methodsOf(generatedClass);
```
Comment (skiddykong) [inline generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java]: final

File: generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java [hunk 8]
```diff
@@ -167,9 +168,9 @@ public void shouldGenerateMethodAnnotatedWithHandlesAnnotationForGET() throws Ex
                                 .with(httpActionWithDefaultMapping(GET)
                                         .withResponseTypes("application/vnd.cakeshop.actionabc+json")))
                         .build(),
-                configurationWithBasePackage(BASE_PACKAGE, outputFolder, generatorProperties().withDefaultServiceComponent()));
+                configurationWithBasePackage(BASE_PACKAGE, outputFolder, generatorProperties().withServiceComponentOf("COMMAND_CONTROLLER")));
 
-        Class<?> generatedClass = compiler.compiledClassOf(BASE_PACKAGE, "RemoteEventProcessorMessageContextCakeshopHandlerCommand");
+        Class<?> generatedClass = compiler.compiledClassOf(BASE_PACKAGE, "RemoteCommandController2EventProcessorMessageContextCakeshopHandlerCommand");
```
Comment (skiddykong) [inline generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java]: final

File: generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java [hunk 9]
```diff
@@ -188,9 +189,9 @@ public void shouldGenerateMethodAcceptingEnvelope() throws MalformedURLException
                                 .withRelativeUri("/cakeshop.controller.command")
                                 .with(httpActionWithDefaultMapping(POST, "application/vnd.cakeshop.actionabc+json")))
                         .build(),
-                configurationWithBasePackage(BASE_PACKAGE, outputFolder, generatorProperties().withDefaultServiceComponent()));
+                configurationWithBasePackage(BASE_PACKAGE, outputFolder, generatorProperties().withServiceComponentOf("COMMAND_CONTROLLER")));
 
-        Class<?> clazz = compiler.compiledClassOf(BASE_PACKAGE, "RemoteEventProcessorMessageContextCakeshopControllerCommand");
+        Class<?> clazz = compiler.compiledClassOf(BASE_PACKAGE, "RemoteCommandController2EventProcessorMessageContextCakeshopControllerCommand");
```
Comment (skiddykong) [inline generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java]: final

File: generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java [hunk 10]
```diff
@@ -188,9 +189,9 @@ public void shouldGenerateMethodAcceptingEnvelope() throws MalformedURLException
...
         Method method = firstMethodOf(clazz);
```
Comment (skiddykong) [inline generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java]: final

Comment (jaceko) [inline generators/messaging-client-generator/src/test/java/uk/gov/justice/services/clients/messaging/generator/MessagingClientGenerator_CodeStructureTest.java]: all fixed

**Pull request merged**